File: app/src/main/java/com/example/kineduexample/ui/fragments/activities/ActivitiesPresenter.java
package com.example.kineduexample.ui.fragments.activities;

import com.example.kineduexample.data.network.KineduInteractor;
import com.example.kineduexample.data.network.model.ActivitiesIndex;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ActivitiesPresenter {
    private KineduInteractor interactor;
    private ActivitiesView view;

    public ActivitiesPresenter(KineduInteractor interactor){
        this.interactor = interactor;
    }

    public void bind(ActivitiesView view){
        this.view = view;
    }

    public void unbind(){
        view = null;
    }

    public void searchActivities(){
        interactor.getActivities()
                .enqueue(new Callback<ActivitiesIndex>() {
                    @Override
                    public void onResponse(Call<ActivitiesIndex> call, Response<ActivitiesIndex> response) {
                        if(view != null){
                            view.onLoadActivities(response.body().getDataActivities().getActivities());
                        }
                    }

                    @Override
                    public void onFailure(Call<ActivitiesIndex> call, Throwable t) {
                        t.printStackTrace();
                    }
                });
    }

}

File: app/src/main/java/com/example/kineduexample/ui/fragments/activities/ActivitiesPresenter.kt
package com.example.kineduexample.ui.fragments.activities

import com.example.kineduexample.data.network.KineduInteractor
import com.example.kineduexample.data.network.model.ActivitiesIndex
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ActivitiesPresenter(private val interactor: KineduInteractor) {
    private var view: ActivitiesView? = null

    fun bind(view: ActivitiesView) {
        this.view = view
    }

    fun unbind() {
        view = null
    }

    fun searchActivities() {
        interactor.getActivities()
                .enqueue(object : Callback<ActivitiesIndex> {
                    override fun onResponse(call: Call<ActivitiesIndex>, response: Response<ActivitiesIndex>) {
                        if (view != null) {
                            view!!.onLoadActivities(response.body().dataActivities!!.activities!!)
                        }
                    }

                    override fun onFailure(call: Call<ActivitiesIndex>, t: Throwable) {
                        t.printStackTrace()
                    }
                })
    }

}
